bangc-ops/kernels/ml_nms/ml_nms_block.mlu

```diff
@@ -0,0 +1,182 @@
+/*************************************************************************
+	Copyright (C) [2022] by Cambricon, Inc.
+    > File Name: main.cpp
+	> Author: wenzhengyin
+	> Mail: jones980116@163.com 
+	> Created Time: Tue Apr 19 14:35:06 2022
+ ************************************************************************/
+#include "bang.h"
+#include "mlu_op_kernel.h"
+#include "kernels/unary_op/unary_op_block.h"
+
+#define MLU_VERSION 270
+#if MLU_VERSION == 220
+#undef MLU_VERSION
+#define MLU_VERSION 220
+#elif MLU_VERSION == 270
+#undef MLU_VERSION
+#define MLU_VERSION 270
+#elif MLU_VERSION == 290
+#undef MLU_VERSION
+#define MLU_VERSION 290
+#elif MLU_VERSION == 322
+#undef MLU_VERSION
+#define MLU_VERSION 322
+#elif MLU_VERSION == 372
+#undef MLU_VERSION
+#define MLU_VERSION 372
+#endif
+
+__mlu_func__ void setSetoff(mluOpDataType_t data_type, int input_boxes_num,
+  int min_cell, int* setoff) {
+  switch (data_type) {
+    case MLUOP_DTYPE_HALF:
+      if ((input_boxes_num % (min_cell / 2)) != 0) {
+        *setoff = (min_cell / 2) * (input_boxes_num / (min_cell / 2) + 1);
+      } else {
+        *setoff = (min_cell / 2) * (input_boxes_num / (min_cell / 2));
+      }
+    case MLUOP_DTYPE_FLOAT:
+      if ((input_boxes_num % (min_cell / 4)) != 0) {
+        *setoff = (min_cell / 4) * (input_boxes_num / (min_cell / 4) + 1);
+      } else {
+        *setoff = (min_cell / 4) * (input_boxes_num / (min_cell / 4));
+      }
+    default:break;
+  }
+}
+
+__mlu_func__ void getOffsetNumMlNmsFast(int input_boxes_num,
+  mluOpDataType_t data_type, int* setoff) {
+  if (MLU_VERSION < 200) {
```

Comment:
If you want to run different function on different devices, you can use `__BANG_ARCH__`. For example, `#if __BANG_ARCH__ < 200`.